app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,41 @@ def show
     @post = Post.find(params[:id])
     @comment = Comment.where(post_id: @post.id).order(created_at: :desc)
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    respond_to do |format|
+      if @post.save
+        format.html { redirect_to user_posts_path }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def comment
+    @post = Post.find(params[:id])
+    @comment = Comment.new(author_id: current_user.id, post_id: @post.id, text: params[:comment])
+    if @comment.save
+      redirect_to user_post_path(current_user)
+    else
+      render :show, alert: 'Error occured!'
+    end
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    Like.new(author_id: current_user.id, post_id: @post.id).save
+    redirect_to user_post_path(current_user)
+  end
```

Comment:
- [ ] Based on the requirements of this project, you are required to have a controller for `comments` and `likes`. The comments controller should have `new` and `create` actions hence let's fix this.
- [ ] Again, there should be a form in the view/comments/new.html.erb file which accepts input for creating comments.

Reply:
Thanks Ill fix it